### PR TITLE
[WIP] to use existing inventory

### LINKED ISSUE #73: use existing inventory during deployments
We always generate new ~/.infrared/workspaces/active/hosts file through [1] when we run the playbook "ansible-playbook main.yml"

Instead, if inventory already exists there, validate the inventory and reuse that. So we need ansible tasks for the below steps
1) read and parse ~/.infrared/workspaces/active/hosts file
may be we can use ansible tasks like below
- name: read hosts file
  shell: |
       ansible-inventory -i ~/.infrared/.workspaces/active/hosts --list -y
  changed_when: false
  register: bulk_status
- name: set fact for introspected nodes
  set_fact
      inventory_fromfile: '{{ bulk_status.stdout | from_yaml }}' 
2) check if the undercloud which we get from instackenv i.e {{ undercloud_hostname }} exists in above inventory i.e {{ inventory_fromfile }}
3) if exists, then read groups and hosts from  {{ inventory_fromfile }} and use those values inside [1] 
- name: add undercloud_host to inventory file
  add_host:
      name: "{{ host.name }}"
      groups: {{ host.groups }}
      ansible_host: {{ host.hostname }}
      ansible_user: {{ host.user }}
      ansible_ssh_private_key_file: {{ host.key }}
      ansible_python_interpreter:  {{ host. python_interpreter }}
 with_dict/items: < hosts read from the file>
- name: generate inventory file
  template:
      dest: "{{ infrared_hosts_dir }}/{{ inventory_file_name }}"
      src: inventory.j2
- name: update inventory file symlink
  file:
      dest: "{{ infrared_hosts_file }}"
      state: link
      src: "{{ inventory_file_name }}"
3) If not exist, prepare new inventory the way we are doing now.

[1] https://github.com/redhat-performance/jetpack/blob/master/tasks/prepare_inventory.yml

### PR DESCRIPTION
**Pull request by asyedham**
Fixes #73

### CODE REVIEW

File: tasks/in_memory_inventory.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+---
+- hosts: localhost
+  tasks:
+    - name: read from hosts file
+      shell: |
+        ansible-inventory -i ~/.infrared/.workspaces/active/hosts --list
```
Comment (smalleni) [inline tasks/in_memory_inventory.yml:6]: Uase {{ ansible_user_dir }}

**Comment by smalleni** [comment]
@venkataanil Please review and merge if needed.

**Comment by smalleni** [comment]
@asyedham @venkataanil is this still relevant?

**Comment by venkataanil** [comment]
It's not relevant for now. But Asma has done good job here. I want her to save these changes locally and push later if we feel required.

**Pull request closed without merging**